sockeye/training.py

```diff
@@ -94,7 +94,7 @@ def _build_module(self,
         Initializes model components, creates training symbol and module, and binds it.
         """
         source = mx.sym.Variable(C.SOURCE_NAME)
-        source_length = mx.sym.Variable(C.SOURCE_LENGTH_NAME)
+        source_length = mx.sym.sum(mx.sym.broadcast_not_equal(source, mx.sym.zeros((1,))), axis=1)
```

Comment:
how about adding an:
```
assert C.PAD_ID == 0
```
to catch issues if we ever changed the PAD_ID and to make it explicit that this is an assumption.